Doc/tutorial/datastructures.rst

```diff
@@ -497,7 +497,8 @@ You can't use lists as keys, since lists can be modified in place using index
 assignments, slice assignments, or methods like :meth:`append` and
 :meth:`extend`.
 
-It is best to think of a dictionary as an unordered set of *key: value* pairs,
+Dictionaries are guaranteed to retain insertion order.
+It is best to think of a dictionary as an ordered set of *key: value* pairs,
```

Comment:
`dictionary as an ordered set` can be misindestood as ordered by key, intead of ordered by insertion order.
It best to write here, that dict will be enumerated  (.keys(), items(), .values()) in order of insertion. And also specify "not ordered by key".

C++ has orderdmap and unorderedmap. Orderedmap orders by key.

Reply:
Agree, thanks.

Reply:
I don’t know if other languages are relevant here.  If Python docs make a clear explanation of sorted and ordered then things should not be confusing.